src/content/en/updates/2018/01/paintapi.md

```diff
@@ -0,0 +1,278 @@
+will be re-drawn on demand, whenever the user resizes the textarea. This means
+the background image is always exactly as big as it needs to be, including the
+compensation for high-density display.
+
+That’s pretty cool, but it’s also quite static. Would we want to write a new
+worklet every time we wanted the same pattern but with differently sized
+squares? The answer is no!
+
+### Parameterizing your worklet
+
+Luckily, the paint worklet can access other CSS properties, which is where the
+additional parameter `properties` comes into play:
+
+```
+<!-- index.html -->
+<!doctype html>
+<style>
+  textarea {
+    /* The paint worklet subscribes to changes of these custom properties*/
+    --checkerboard-spacing: 10;
+    --checkerboard-size: 32;
+    background-image: paint(demo);
+  }
+</style>
+<textarea></textarea>
+<script>
+  CSS.paintWorklet.addModule('checkerboard.js');
+</script>
+```
+
+```
+// checkerboard.js
+class CheckerboardPainter {
+  // inputProperties returns a list of CSS properties that this paint function gets access to
+  static get inputProperties() { return ['--checkerboard-spacing', '--checkerboard-size']; }
+
+  paint(ctx, geom, properties, args) {
+    // Paint Worklet uses CSS Typed OM to model the input values. As of now, they are mostly wrappers around strings, but will be augmented to hold more accessible data over time.
+    const size = parseInt(properties.get('--checkerboard-size').toString());
+    const spacing = parseInt(properties.get('--checkerboard-spacing').toString());
+    const colors = ['red', 'green', 'blue'];
+    for(let y = 0; y < geom.height/size; y++) {
+      for(let x = 0; x < geom.width/size; x++) {
+        ctx.fillStyle = colors[(x + y) % colors.length];
+        ctx.beginPath();
+        ctx.rect(x*(size + spacing), y*(size + spacing), size, size);
+        ctx.fill();
+      }
+    }
+  }
+}
+
+registerPaint('checkerboard', CheckerboardPainter);
+```
+
+Now we can use the same code for all different kind of checkerboards. But even
+better, we can now go into DevTools and [fiddle with the
+values](https://googlechromelabs.github.io/houdini-samples/paint-worklet/parameter-checkerboard/)
+until we find the right look.
+
+<div style="display: flex; justify-content: center">
+  <video loop muted controls>
+    <source
+      src="https://storage.googleapis.com/webfundamentals-assets/paintapi/checkercast_vp8.webm"
+      type="video/webm; codecs=vp8">
+    <source
+      src="https://storage.googleapis.com/webfundamentals-assets/paintapi/checkercast_x264.mp4"
+      type="video/mp4; codecs=h264">
+  </video>
+</div>
+
+Note: It would be great to parameterize the colors, too, wouldn’t it? The spec
+allows for the `paint()` function to take a list of arguments. This feature is
+not implemented in Chrome yet, as it heavily relies on Houdini’s Properties &
+Values API, which still needs some work before it can ship.
+
+## Browsers that doesn’t support paint worklet
```

Comment:
"Browser support"

Reply:
+1 

I often forget to look at the headings. This one has a doozy of a grammar mistake.